### PR TITLE
Feature/checkins

### PR DESCRIPTION
**Pull request by ryanrampage1**

### CODE REVIEW

File: app/src/main/java/com/spartahack/spartahack17/Retrofit/CheckInResponse.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.spartahack.spartahack17.Model.Session;
+
+import java.util.ArrayList;
+
+/**
+ * Created by ryancasler on 1/16/17
+ * Spartahack-Android
+ */
+public class CheckInResponse extends Session{
```
Comment (memuyskens) [inline app/src/main/java/com/spartahack/spartahack17/Retrofit/CheckInResponse.java]: Might want to add a space after Session.

File: app/src/main/res/layout/activity_check_in.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        </android.support.v7.widget.Toolbar>
+
+    </android.support.design.widget.AppBarLayout>
+
+    <Button
+        android:id="@+id/scan"
+        style="@style/ButtonStyle"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="10dp"
```
Comment (memuyskens) [inline app/src/main/res/layout/activity_check_in.xml]: For the sake of consistency, you might consider using the vertical margin in dimensions xml.

File: app/src/main/res/layout/activity_check_in.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        android:text="@string/start_scan" />
+
+    <TextView
+        android:id="@+id/result"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="@dimen/activity_vertical_margin"
+        android:gravity="center"
+        android:textSize="@dimen/header_text_size"
+        tools:text="Here is the rusult of the scan" />
```
Comment (memuyskens) [inline app/src/main/res/layout/activity_check_in.xml:72]: This should be extracted to strings xml

File: app/src/main/res/layout/activity_login.xml [hunk 4]
```diff
@@ -0,0 +1,208 @@
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (memuyskens) [inline app/src/main/res/layout/activity_login.xml]: Is this file even used? I didn't find usages anywhere. It also has the push notifications switches within it and those were moved to settings fragment.

File: app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+/**
+ * Created by ryancasler on 1/14/17
+ * Spartahack-Android
+ */
+public class CheckinActivity extends BaseActivity {
+    public static final int UNPROCESSABLE_ENTITY = 422;
+
+    @BindView(R.id.result) TextView result;
+//    @BindView(R.id.progress_bar) ProgressBar progressBar;
```
Comment (memuyskens) [inline app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java]: Will we be using this progress bar?

File: app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                    } else {
+                        showError(null);
+                    }
+                });
+    }
+
+    private void minorCheckIn() {
+        new AlertDialog.Builder(this)
+                .setTitle("Minor Check In")
+                .setMessage("This user is a minor do you have all their forms? ")
```
Comment (memuyskens) [inline app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java]: I'd move this to strings xml

File: app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                .setPositiveButton("Check In", (dialog, which) -> {
```
Comment (memuyskens) [inline app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java]: I think there's already a string in the xml that says "Check In"?

File: app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+                    } else {
+                        showError(null);
+                    }
+                });
+    }
+
+    private void minorCheckIn() {
+        new AlertDialog.Builder(this)
+                .setTitle("Minor Check In")
```
Comment (memuyskens) [inline app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java]: I'd move this to strings xml

**Review by memuyskens — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+                })
+                .show();
+    }
+
+    private void showError(String error) {
+        currentScanId = -1;
+        result.setText(TextUtils.isEmpty(error) ? "Error Checking User In" : error);
+    }
+
+    @OnClick(R.id.scan) public void scan(){
```
Comment (memuyskens) [inline app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java]: Space after scan()

File: app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java (hunk 5 above)
Comment (ryanrampage1) [reply app/src/main/java/com/spartahack/spartahack17/Activity/CheckinActivity.java]: I removed it

File: app/src/main/res/layout/activity_check_in.xml (hunk 3 above)
Comment (ryanrampage1) [reply app/src/main/res/layout/activity_check_in.xml:72]: I used the tools:text to avoid this problem.

Comment (memuyskens) [reply app/src/main/res/layout/activity_check_in.xml:72]: Ahhhhh thanks for clarifying.

**Review by memuyskens — APPROVED** [review]

**Pull request merged**